### DISCUSSION TITLE
TE gives a warning with relationship even though one is active and one inactive

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1268] [Data Modeling]
I have Fact and Dim   Cust_ID in both
I have one relationship with MANY on fact and one on DIM  by Cust_ID ONE DIRECTION set to ACTIVE
I wanted to add another relationship with MANY on fact and one on DIM by Cust_ID set to BOTH directions set to INACTIVE

I have cases where i need to see EVERYTHING in DIM even if they have no FACTS. 
Why is TE gives a warning with relationship even though one is active and one inactive
Is this not allowed

Case: One to Many
So if i have a visual as a matrix with 
measure TOTAL CUSTOMERS (from the customer table) 
row 1 Y exists in fact table N does not
row 2 a Year linked from date table to fact table
Output is all the same TOTAL Customers for each year (not desired)

Case: Both directions
So if i have a visual as a matrix with 
measure TOTAL CUSTOMERS (from the customer table) 
row 1 Y exists in fact table N does not
row 2 a Year linked from date table to fact table
Output is the correct TOTAL Customers for each year (perfect)

But there are cases where when using both i can get ambigous errors.
This is why i want the default to be one to many, and invoke BOTH for certain scenarios.

The only other thing i think of is have a duplicate copy of dim Custormer and set that to BOTH 
But that can't be the preferred method, particular with 20 M customers

**Comment by mlonsk** [comment]
Hi @TinyTimothy2 
I do not believe that semantic models allow for multiple relationships to the same column even if one is inactive. 
The way to handle this is probably to have another Cust_ID in your fact table and then create the inactive bidirectional relationship on that.

**Reply by TinyTimothy2** [reply]
Thx - have you ever tried this? I am surprised there is not any discussions on this.  The obvious downside is another high cardinality field in the fact table

**Comment by otykier** [comment]
You don't need to set up multiple relationships between the same two columns to achieve this (which isn't possible, as @mlonsk pointed out). Simply use the DAX [`CROSSFILTER`](https://dax.guide/crossfilter) function, which lets you specify the type of filtering to apply (e.g. 'OneWay', 'Both', etc.).
